toolz/itertoolz.py

```diff
@@ -9,14 +9,27 @@
 from toolz.utils import no_default
 
 
-__all__ = ('remove', 'accumulate', 'groupby', 'merge_sorted', 'interleave',
-           'unique', 'isiterable', 'isdistinct', 'take', 'drop', 'take_nth',
-           'first', 'second', 'nth', 'last', 'get', 'concat', 'concatv',
-           'mapcat', 'cons', 'interpose', 'frequencies', 'reduceby', 'iterate',
-           'sliding_window', 'partition', 'partition_all', 'count', 'pluck',
+__all__ = ('empty', 'remove', 'accumulate', 'groupby', 'merge_sorted',
+           'interleave', 'unique', 'isiterable', 'isdistinct', 'take',
+           'drop', 'take_nth', 'first', 'second', 'nth', 'last', 'get',
+           'concat', 'concatv', 'mapcat', 'cons', 'interpose',
+           'frequencies', 'reduceby', 'iterate', 'sliding_window',
+           'partition', 'partition_all', 'count', 'pluck',
            'join', 'tail', 'diff', 'topk', 'peek', 'random_sample')
 
 
+def empty():
```

Comment:
Yeah that would work too. Was kind of wanting it to be a `generator` type. Though if there is a strong preference to change, it we can do that.